driver/avgpool_driver.hpp

```diff
@@ -0,0 +1,595 @@
+    count_include_pad = static_cast<bool>(inflags.GetValueInt("count_include_pad"));
+    divisor_override  = inflags.GetValueInt("divisor_override");
+
+    N = in_dim[0];
+    C = in_dim[1];
+    D = in_dim.size() == 5 ? in_dim[2] : 1;
+    H = in_dim.size() == 5 ? in_dim[3] : in_dim[2];
+    W = in_dim.size() == 5 ? in_dim[4] : in_dim[3];
+
+    std::vector<int64_t> out_dim;
+    if(in_dim.size() == 5)
+    {
+        if(ceil_mode)
+        {
+            OD = std::ceil(static_cast<float>(D - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+            OH = std::ceil(static_cast<float>(H - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+            OW = std::ceil(static_cast<float>(W - ksize[2] + 2 * padding[2]) / stride[2]) + 1;
+        }
+        else
+        {
+            OD = std::floor(static_cast<float>(D - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+            OH = std::floor(static_cast<float>(H - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+            OW = std::floor(static_cast<float>(W - ksize[2] + 2 * padding[2]) / stride[2]) + 1;
+        }
+        out_dim = std::vector<int64_t>{N, C, OD, OH, OW};
+    }
+    else
+    {
+        if(ceil_mode)
+        {
+            OH = std::ceil(static_cast<float>(H - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+            OW = std::ceil(static_cast<float>(W - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+        }
+        else
+        {
+            OH = std::floor(static_cast<float>(H - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+            OW = std::floor(static_cast<float>(W - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+        }
+        out_dim = std::vector<int64_t>{N, C, OH, OW};
+    }
+    std::vector<int> out_grad_stride = ComputeStrides(out_dim);
+    SetTensorNd(inputDesc, in_dim, in_stride, data_type);
+    SetTensorNd(outputDesc, out_dim, data_type);
+    SetTensorNd(outputGradDesc, out_dim, out_grad_stride, data_type);
+    SetTensorNd(inputGradDesc, in_dim, data_type);
+
+    return miopenStatusSuccess;
+}
+
+// Equivalent to: tensor.tranpose(0, -1).contiguous().tranpose(0, -1) incase contiguous = False
+template <typename Tgpu, typename Tref>
+std::vector<int> AvgPoolDriver<Tgpu, Tref>::ComputeStrides(std::vector<int> inputDim)
+{
+    if(!isContiguous)
+        std::swap(inputDim.front(), inputDim.back());
+    std::vector<int> strides(inputDim.size());
+    strides.back() = 1;
+    for(int i = inputDim.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * inputDim[i + 1];
+    if(!isContiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+int AvgPoolDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward AvgPool (Default=1)", "int");
+    inflags.AddInputFlag(
+        "input_dims",
+        'D',
+        "2,3,7,9",
+        "The dimensional lengths of the input tensor: N,C,D1,D2,... Example: 2,3,7,9.",
+        "string");
```

Comment:
use AddTensorFlag to read tensor shape from command line

Reply:
Is this from AMD's demand?, if not, it is accepted by Moreh's engineer

Reply:
I agree with @cognaiger9. Our recently PRs in ROCm/MIOpen are forced to use `AddTensorFlag` from AMD.

Reply:
Added it